servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -9984,16 +10017,35 @@ public void run() {
     }
 
     public void recordMetrics(ResourceContext ctx, int httpStatus) {
-        final String principalDomainName = getPrincipalDomain(ctx);
-        final String domainName = getRequestDomainName(ctx);
-        final Object timerMetric = getTimerMetric(ctx);
-        final String httpMethod = (ctx != null) ? ctx.getHttpMethod() : null;
-        final String apiName = (ctx != null) ? ctx.getApiName() : null;
-        final String timerName = (apiName != null) ? apiName + "_timing" : null;
-        metric.increment("zms_api", domainName, principalDomainName, httpMethod, httpStatus, apiName);
-        metric.stopTiming(timerMetric, domainName, principalDomainName, httpMethod, httpStatus, timerName);
+        try {
+            final String principalDomainName = getPrincipalDomain(ctx);
+            final String domainName = getRequestDomainName(ctx);
+            final Object timerMetric = getTimerMetric(ctx);
+            final String httpMethod = (ctx != null) ? ctx.getHttpMethod() : null;
+            final String apiName = (ctx != null) ? ctx.getApiName() : null;
+            final String timerName = (apiName != null) ? apiName + "_timing" : null;
+            metric.increment("zms_api", domainName, principalDomainName, httpMethod, httpStatus, apiName);
+            metric.stopTiming(timerMetric, domainName, principalDomainName, httpMethod, httpStatus, timerName);
+        } catch (Exception e) {
+            LOG.warn("Got exception during recordMetrics: {}", e.getMessage(), e);
+        }
     }
 
+    @Override
+    public void publishChangeMessage(ResourceContext ctx, int httpStatus) {
+        try {
+            if (httpStatus == 200) {
+                for (DomainChangeMessage changeMessage : ctx.getDomainChangeMessages()) {
+                    for (ChangePublisher<DomainChangeMessage> publisher : domainChangePublishers) {
```

Comment:
if our configured topic names is empty, then we'll end up having a null domainChangePublishers which will cause null pointer exception so we should check if it's null or not at the beginning of the method.

Reply:
added null check in the creation instead.

Reply:
I was actually referring to the domainChangePublishers object itself. If the topic property setting is null or empty string, then domainChangePublishers itself is null. It only gets initialized if we have at least one entry. So at the beginning of the method we can just check 

if (domainChangePublishers == null) {
   return;
}

Reply:
missed that ,thanks
fixed